packages/nci-openmpi/package.py

```diff
@@ -0,0 +1,59 @@
+## Adapted from https://github.com/spack/spack/blob/v0.20.0/var/spack/repos/builtin/packages/openmpi/package.py
+## Apache-2.0 Licence
+## Scott Wales 2023
+
+
+class NciOpenmpi(Package):
+    """
+    Represents NCI's install of openmpi, where GCC and Intel variants of the
+    library are stored under lib/GNU and lib/Intel respectively
+
+    Not an installable package, this should instead be added to packages.yaml as::
+
+        nci-openmpi:
+          externals:
+            - spec: nci-openmpi@4.1.4
+              prefix: /apps/openmpi/4.1.4
+          buildable: False
```

Comment:
Was that to fix an error or a stylistic choice?

Reply:
I added it before starting my tests. I began doing it because it sets up the environment variables with the path to the relevant files: https://github.com/ACCESS-NRI/ACCESS-OM/issues/6#issuecomment-1594184659

Reply:
Just did a test without the automatic module load. ACCESS-OM2 built successfully.

Reply:
The module load changes quite a lot of the environment:
```
$ module display openmpi/4.1.4
-------------------------------------------------------------------
/apps/Modules/modulefiles/openmpi/4.1.4:

setenv          HCOLL_ENABLE_MCAST 0
conflict        mpi lam mpich openmpi intel-mpi
setenv          OMPI_BASE /apps/openmpi/4.1.4
setenv          OMPI_ROOT /apps/openmpi/4.1.4
setenv          OMPI_VERSION 4.1.4
setenv          VT_MAX_FLUSHES 0
setenv          VT_PFORM_LDIR /scratch/tm70/aph502/tmp
setenv          OMPI_MCA_orte_tmpdir_base /tmp
prepend-path    PATH /apps/openmpi/4.1.4/bin
prepend-path    C_INCLUDE_PATH /apps/openmpi/4.1.4/include
prepend-path    CPLUS_INCLUDE_PATH /apps/openmpi/4.1.4/include
prepend-path    CPATH /apps/openmpi/4.1.4/include
prepend-path    FPATH /apps/openmpi/4.1.4/include
prepend-path    LIBRARY_PATH /apps/openmpi/4.1.4/lib/profilers
prepend-path    LD_LIBRARY_PATH /apps/openmpi/4.1.4/lib/profilers
prepend-path    LD_RUN_PATH /apps/openmpi/4.1.4/lib/profilers
prepend-path    LIBRARY_PATH /apps/openmpi/4.1.4/lib
prepend-path    LD_LIBRARY_PATH /apps/openmpi/4.1.4/lib
prepend-path    LD_RUN_PATH /apps/openmpi/4.1.4/lib
prepend-path    MANPATH /apps/openmpi/4.1.4/share/man
setenv          UCC_LOG_LEVEL ERROR
setenv          UCC_TLS nccl,ucp
prepend-path    PATH /apps/openmpi/wrapper
prepend-path    PATH /apps/openmpi/wrapper/fortran
setenv          OPENMPI_BASE /apps/openmpi/4.1.4
setenv          OPENMPI_ROOT /apps/openmpi/4.1.4
setenv          OPENMPI_VERSION 4.1.4
module-whatis   {openmpi, version 4.1.4}
-------------------------------------------------------------------
```
It isn't clear to me that a test of the ACCESS-OM2 build depends on all of those modifications: there could be use cases that benefit from the module load. In which case I think it is better to add it.